src/main/java/it/pagopa/pm/gateway/service/CcResumeStep2Service.java

```diff
@@ -0,0 +1,182 @@
+            }
+        } catch (Exception e) {
+            log.error("{}{}", GENERIC_ERROR_MSG, entity.getIdTransaction(), e);
+        }
+    }
+
+    private HttpClientResponse callVPos(Map<String, String> params) throws IOException {
+        HttpClientResponse clientResponse = httpClient.post(vposUrl, ContentType.APPLICATION_FORM_URLENCODED.getMimeType(), params);
+        if (clientResponse.getStatus() != HttpStatus.OK.value()) {
+            log.error("HTTP Response Status: {}", clientResponse.getStatus());
+            throw new IOException("Non-ok response from VPos. HTTP status: " + clientResponse.getStatus());
+        }
+        return clientResponse;
+    }
+
+    private boolean isStepTwoResultCodeOk(ThreeDS2Response response, PaymentRequestEntity entity) {
+        String resultCode = response.getResultCode();
+        String status = CREATED.name();
+        String responseType = StringUtils.EMPTY;
+        String correlationId = StringUtils.EMPTY;
+        boolean isToAccount = false;
+        if (RESULT_CODE_AUTHORIZED.equals(resultCode)) {
+            responseType = response.getResponseType().name();
+            isToAccount = true;
+            correlationId = ((ThreeDS2Authorization) response.getThreeDS2ResponseElement()).getTransactionId();
+        } else {
+            log.error("Error resultCode {} from Vpos for requestId {}", resultCode, entity.getGuid());
+            status = DENIED.name();
+        }
+        entity.setCorrelationId(correlationId);
+        entity.setStatus(status);
+        entity.setResponseType(responseType);
+        paymentRequestRepository.save(entity);
+        return isToAccount;
+    }
+
+    private void executeAccount(PaymentRequestEntity entity, StepZeroRequest pgsRequest) {
+        try {
+            log.info("Calling VPOS - Accounting - for requestId: {}", entity.getGuid());
+            Map<String, String> params = vPosRequestUtils.buildAccountingRequestParams(pgsRequest, entity.getCorrelationId());
+            HttpClientResponse clientResponse = callVPos(params);
+            AuthResponse response = vPosResponseUtils.buildAuthResponse(clientResponse.getEntity());
+            vPosResponseUtils.validateResponseMac(response.getTimestamp(), response.getResultCode(), response.getResultMac(), pgsRequest);
+            checkAccountResultCode(response, entity);
+        } catch (Exception e) {
+            log.error(GENERIC_ERROR_MSG + entity.getIdTransaction() + " stackTrace: " + Arrays.toString(e.getStackTrace()));
+        }
+        executePatchTransaction(entity);
+    }
+
+    private void checkAccountResultCode(AuthResponse response, PaymentRequestEntity entity) {
+        String resultCode = response.getResultCode();
+        String status = AUTHORIZED.name();
+        boolean authorizationOutcome = true;
+        if (!resultCode.equals(RESULT_CODE_AUTHORIZED)) {
+            status = DENIED.name();
+            authorizationOutcome = false;
+        }
```

Comment:
```suggestion
        boolean authorizationOutcome = resultCode.equals(RESULT_CODE_AUTHORIZED);
        if (!authorizationOutcome) {
            status = DENIED.name();
        }
```

Reply:
Addressed in [63c1344](https://github.com/pagopa/pagopa-payment-transactions-gateway/pull/189/commits/63c1344951ba906fc72c75e3aa92cd57e915c7ed)